File: backend/BeautyCommerce.Domain/Entities/Product.cs
using BeautyCommerce.Domain.Base;
using BeautyCommerce.Domain.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;

    public string Slug { get; set; } = string.Empty;

    public string ShortDescription { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public Guid BrandId { get; set; }

    public Brand Brand { get; set; } = null!;

    public Guid CategoryId { get; set; }

    public Category Category { get; set; } = null!;

    public bool IsFeatured { get; set; }

    public bool IsNew { get; set; }

    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();

    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();

    public ICollection<WishlistItem> WishlistItems { get; set; } = new List<WishlistItem>();
}

File: frontend/web/src/types/product.ts
export interface Product {
  id: string;
  sku: string;
  barcode: string;

  name: string;
  slug: string;

  shortDescription?: string;
  description?: string;

  price: number;
  oldPrice?: number;

  brandId: string;
  categoryId: string;

  brand?: {
    id: string;
    name: string;
  };

  category?: {
    id: string;
    name: string;
  };

  images?: ProductImage[];

  isFeatured?: boolean;

  variants?: ProductVariant[];
  averageRating?: number;
  reviewCount?: number;
  defaultVariantId?: string;
}

export interface ProductImage {
  id: string;
  imageUrl: string;
  isPrimary?: boolean;
  displayOrder?: number;
}

export interface ProductVariant {
  id: string;
  productId: string;
  sku: string;
  color?: string;
  size?: string;
  stock: number;
  price: number;
}
